pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+
+	isRetryableKey = tag.MustNewKey("is_retryable")
+)
+
+type workflowMetrics struct {
+	// Total successful workflow operation requests
+	workflowOperationsTotal *stats.Int64Measure
+	// Total failed workflow operation requests
+	workflowOperationsFailedTotal *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+
+	// Total workflow/activity reminders created
+	workflowRemindersTotal *stats.Int64Measure
+
+	// Total successful workflow/activity executions
+	workflowExecutionTotal *stats.Int64Measure
+	// Total failed workflow/activity executions
+	workflowExecutionFailedTotal *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+
+	// Latency between execution request and actual execution
+	workflowSchedulingLatency *stats.Float64Measure
+
+	appID   string
+	enabled bool
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationsTotal: stats.Int64(
+			"runtime/workflow/operations/total",
```

Comment:
Any reason behind adding `runtime` prefix? That doesn't seem to be adding much information

Reply:
yes, actually I tried to name them as suggested in the issue itself here

https://github.com/dapr/dapr/issues/7109

Reply:
Oh okay. Yeah we can keep it as well. No issues

Reply:
Sure.